### ISSUE TITLE
Error 500 on Category and Monitoring on 8.01 version

### ISSUE THREAD

**Issue by darekcza** [PrestaShop/PrestaShop#31075]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I just did the migration from version 1.6x to version 8.01 using Migration 4.0 module by ETS-Soft
When I click Category or Monitoring in the back office I get 500 error, please help, here is the stack trace from category click:

```
TypeError:
stripslashes() expects parameter 1 to be string, null given

  at src/Core/Grid/Data/Factory/AbstractCategoryDataFactory.php:77
  at stripslashes(null)
       (src/Core/Grid/Data/Factory/AbstractCategoryDataFactory.php:62)
  at PrestaShop\PrestaShop\Core\Grid\Data\Factory\AbstractCategoryDataFactory->getData(object(CategoryFilters))
     (src/Core/Grid/GridFactory.php:90)
  at PrestaShop\PrestaShop\Core\Grid\GridFactory->getGrid(object(CategoryFilters))
     (src/Core/Grid/Factory/CategoryGridFactoryDecorator.php:72)
  at PrestaShop\PrestaShop\Core\Grid\Factory\CategoryGridFactoryDecorator->getGrid(object(CategoryFilters))
     (src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:105)
  at PrestaShopBundle\Controller\Admin\Sell\Catalog\CategoryController->indexAction(object(Request), object(CategoryFilters))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:169)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:81)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, false)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request), 1, false)

```

### Steps to reproduce

click Categories in the back office or Monitoring

### PrestaShop version(s) where the bug happened

8.01

**Comment by darekcza** [comment]
I found the problem!
in category_lang table there were NULL records showing in sql (phpMyAdmin), for example:
`(1, 1, 1, 'Bazowa', NULL, NULL, 'bazowa', NULL, NULL, NULL),`
I changed Null to '' like this:
`(1, 1, 1, 'Bazowa', '', '', 'bazowa', '', '', ''),`

now both categories and monitoring work

**Comment by hibatallahAouadni** [comment]
Hello @darekcza 

I didn't manage to reproduce the issue with PS8.0.1. Did you upgrade your shop lately? 
If so, what was your previous version? and which module did you use to upgarde? and what was the module version?
Please check and feedback.

Thanks!

**Comment by SalvatoreCervone** [comment]
i have same problem.... but replace null to '' not resolve

**Comment by florine2623** [comment]
Hello @darekcza ,

If you are using an external module, please contact the develop of the module directly.

If you want to upgrade, you can also use the [autoupgrade module](https://github.com/PrestaShop/autoupgrade/releases).
I advise you to upgrade from 1.6 to [1.7](https://devdocs.prestashop-project.org/1.7/basics/installation/system-requirements/) first using PHP 7.1. And the from 1.7 to [8.0](https://devdocs.prestashop-project.org/8/basics/installation/system-requirements/) using PHP 7.2 or superior versions.

I'll close your issue as it couldn't be reproduce by the team.
Feel free to open a new one if you encounter an issue using the autoupgrade module.

Thanks!

**Comment by nbao** [comment]
Because when you export database in php application like phpmyadmin or adminer or prestashop/admin/configure/advanced/backups/ It will rewrite the empty field in INSERT sql with NULL.
this error comes from code 2 years ago with php 7.2 So developer can't test this error at this moment
![image](https://user-images.githubusercontent.com/5112514/231785373-b758eadd-e2c0-4fa7-bd43-1fcfddf738e1.png)
it fixes with empty check 

if(!empty($record['description'])){
            $records[$key]['description'] = mb_substr(strip_tags(stripslashes($record['description'])), 0, self::DESCRIPTION_MAX_LENGTH);
          }

**Comment by 01Kuzma** [comment]
The same error exists in 8.03 version too!
I've upgraded from 1.7.8.8 with AutoUpgrade module on PHP 7.4.

As @nbao suggests, this code works:

```

        foreach ($records as $key => $record) {
			if(!empty($record['description'])) {
				$records[$key]['description'] = mb_substr(strip_tags(stripslashes($record['description'])), 0, self::DESCRIPTION_MAX_LENGTH);
			}
        }
```

Probably, as a simple PR must be done

**Comment by kmorgen** [comment]
I had similar problem when updating PS 1.7.8.10 to PS 8.1.1 (PHP 8.1) with 1-click Upgrade V4.16.3.
Cannot access several menu points like Categories.

Running this SQL Query (to all replace the NULLs) solved it for me:
UPDATE ps_category_lang SET description = '' WHERE description IS NULL

**Comment by danielle771** [comment]
Hi everyone!

I just installed Prestashop 8.1.6 and I get this HTTP 500 Internal Server Error when I try to create or view a product.

Can anyone help??? (using php 7.4.33, I tried php 8.1.28 but got the same errors)

Thank you in advance. 

![Screenshot 2024-05-17 at 16-28-15 Notice Undefined offset 1 (500 Internal Server Error)](https://github.com/PrestaShop/PrestaShop/assets/170107535/f1a751e1-5328-40e7-8d68-62947dfe5f48)
